modules/Movecraft/src/main/java/net/countercraft/movecraft/mapUpdater/update/CraftTranslateCommand.java

```diff
@@ -178,8 +178,11 @@ public void doUpdate() {
             craft.setProcessing(false);
         time = System.nanoTime() - time;
         if(Settings.Debug)
-            logger.info("Total time: " + (time / 1e9) + " seconds. Moving with cooldown of " + craft.getTickCooldown() + ". Speed of: " + String.format("%.2f", craft.getSpeed()));
-        craft.addMoveTime(time/1e9f);
+            logger.info("Total time: " + (time / 1e6) + " milliseconds. Moving with cooldown of " + craft.getTickCooldown() + ". Speed of: " + String.format("%.2f", craft.getSpeed()) + ". Displacement of: " + displacement);
+
+        // Only add cruise time if cruising
+        if(displacement.getZ() == 0 && displacement.distanceSquared(new MovecraftLocation(0,0,0)) == Math.pow(craft.getType().getCruiseSkipBlocks() + 1, 2))
```

Comment:
You probably meant to use getY, and you can just check if the craft is cruising

Reply:
The goal of this check is to make sure the displacement is exactly the cruise skip blocks +1, which means that it isn't banking or diving.  GetCruising will return true for all cases of banking and diving (since they are only in AsyncManager, and here as displacement).